sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -91,5 +99,67 @@ internal static string GenerateUserAgentString(Assembly clientAssembly, string?
         /// The properly formatted UserAgent string based on this <see cref="TelemetryDetails"/> instance.
         /// </summary>
         public override string ToString() => _userAgent;
+
+        /// <summary>
+        /// If the ProductInformation is not in the proper format, this escapes any ')' , '(' or '\' characters per https://www.rfc-editor.org/rfc/rfc7230#section-3.2.6
+        /// </summary>
+        /// <param name="productInfo">The ProductInfo portion of the UserAgent</param>
+        /// <returns></returns>
+        private static string EscapeProductInformation(string productInfo)
+        {
+            // If the string is already valid, we don't need to escape anything
+            bool success = false;
+            try
+            {
+                success = ProductInfoHeaderValue.TryParse(productInfo, out var _);
+            }
+            catch (Exception)
+            {
+                // Invalid values can throw in Framework due to https://github.com/dotnet/runtime/issues/28558
+                // Treat this as a failure to parse.
+            }
+            if (success)
+            {
+                return productInfo;
+            }
+
+            var sb = new StringBuilder(productInfo.Length + 2);
+            sb.Append('(');
+            // exclude the first and last characters, which are the enclosing parentheses
```

Comment:
But do we know this for sure given that this is an invalid string?

Reply:
Ah nevermind - these are the parens that we have added ourselves.

Reply:
Yes, because we put those enclosing parens there [when calling this](https://github.com/Azure/azure-sdk-for-net/pull/34355/files/1dcaa44b9305801ca1f31dfc2f0f87eb48195569#diff-2b50cdd937e40ec812320781c9d4a8319f24484dde3263c0cd57b4b9428e3525R91)